### PR TITLE
Add a new consumer level event which fires on a rebalance 

### PR DESCRIPTION
**Pull request by benbirt97**
### Description
I have added a `rebalancing` event on the consumer so that applications using KafkaJS have an easy interface they can use in order to be notified a rebalance is going on.

### Reason
Currently there is no simple way to know when a rebalance has occurred. We encountered an issue where we had metrics which tracked partition specific consumer lag. However if the application lost the partition due to a rebalance there was no way to remove the metric, as the app didn't know a rebalance had occurred. This event will allow us to be informed of when a rebalance has occurred so we can reset these metrics within the app.

**See issue #1066**

### Changes
- Added an event emitter for rebalancing on the consumer
- Added test to check that the rebalancing works
- Updated types
- Updated docs

### CODE REVIEW

**Review by Nevon — CHANGES_REQUESTED** [review]
The change looks good overall, just one minor thing to address in the tests.

File: src/consumer/__tests__/instrumentationEvents.spec.js [hunk 1]
```diff
@@ -354,6 +354,56 @@ describe('Consumer > Instrumentation Events', () => {
...
+    consumer2.run({ eachMessage: () => true })
+
+    await waitForConsumerToJoinGroup(consumer2, { label: 'consumer2' })
+
+    expect(onRebalancing).toBeCalledWith({
+      id: expect.any(Number),
+      timestamp: expect.any(Number),
+      type: 'consumer.rebalancing',
+      payload: {
+        groupId: expect.any(String),
```
[the remark this answers is not in the document]
Comment (benbirt97) [reply src/consumer/__tests__/instrumentationEvents.spec.js]: Good call, I've updated the test. I also added a type for the result at the same time 👍

**Comment by benbirt97** [comment]
All tests seemed to work when I ran them, looks like the memberId is being overwritten before it hits the check. Will have a look into this in a bit

File: src/consumer/runner.js [hunk 2]
```diff
@@ -495,6 +500,11 @@ module.exports = class Runner extends EventEmitter {
             retryTime,
           })
 
+          this.instrumentationEmitter.emit(REBALANCING, {
```
[the remark this answers is not in the document]
Comment (Nevon) [reply src/consumer/runner.js:503]: Created an issue for this #1069

**Review by Nevon — APPROVED** [review]

**Pull request merged**